linux/software/libcamera/csi-2-usage.md

```diff
@@ -0,0 +1,166 @@
+
+
+### V4L2
+
+There is a fully open source kernel driver available for the Unicam block; this is a kernel module called bcm2835-unicam. This interfaces to V4L2 subdevice drivers for the source to deliver the raw frames. This bcm2835-unicam driver controls the sensor, and configures the CSI-2 receiver so that the peripheral will write the raw frames (after Debayer) to SDRAM for V4L2 to deliver to applications. Except for this ability to unpack the CSI-2 Bayer formats to 16bits/pixel, there is no image processing between the image source (e.g. camera sensor) and bcm2835-unicam placing the image data in SDRAM.
+
+```
+|------------------------|
+|     bcm2835-unicam     |
+|------------------------|
+     ^             |
+     |      |-------------|
+ img |      |  Subdevice  |
+     |      |-------------|
+     v   -SW/HW-   |
+|---------|   |-----------|
+| Unicam  |   | I2C or SPI|
+|---------|   |-----------|
+csi2/ ^             |
+ccp2  |             |
+    |-----------------|
+    |     sensor      |
+    |-----------------|
+```
+
+Mainline Linux has a range of existing drivers. The Raspberry Pi kernel tree has some additional drivers and device tree overlays to configure them that have all been tested and confirmed to work. They include:
+
+| Device | Type | Notes |
+| :----- | :--- | :---- |
+| Omnivision OV5647 | 5MP Camera | Original Raspberry Pi Camera |
+| Sony IMX219 | 8MP Camera | Revision 2 Raspberry Pi camera |
+| Sony IMX477 | 12MP Camera | Raspberry Pi HQ camera |
+| Toshiba TC358743 | HDMI to CSI-2 bridge | |
+| Analog Devices ADV728x-M | Analog video to CSI-2 bridge| No interlaced support |
+| Infineon IRS1125 | Time-of-flight depth sensor| Supported by a third party |
+
+As the subdevice driver is also a kernel driver, with a standardised API, 3rd parties are free to write their own for any source of their choosing. 
+
+## Developing a third-party driver for bcm2835-unicam
+
+This is the recommended approach to interfacing via Unicam.
+
+When developing a driver for a new device intended to be used with the bcm2835-unicam module, you need the driver and corresponding device tree overlays. Ideally the driver should be submitted to the [linux-media](http://vger.kernel.org/vger-lists.html#linux-media) mailing list for code review and merging into mainline, then moved to the [Raspberry Pi kernel tree](https://github.com/raspberrypi/linux), but exceptions may be made for the driver to be reviewed and merged directly to the Raspberry Pi kernel.
+
+Please note that all kernel drivers are licensed under the GPLv2 license, therefore source code **MUST** be available. Shipping of binary modules only is a violation of the GPLv2 license under which the Linux kernel is licensed. 
```

Comment:
license -> licence :wink: 

(it's spelt with 's' when used as a verb, but spelt with 'c' when used as a noun. E.g. TV Licences are issued by the TV Licensing Authority.)

Reply:
I probably knew that, once.